Core/Core.cpp

```diff
@@ -112,13 +114,15 @@ bool Core_IsInactive() {
 
 void Core_WaitInactive() {
 	while (Core_IsActive()) {
-		m_hInactiveEvent.wait(m_hInactiveMutex);
+		std::unique_lock<std::mutex> guard(m_hInactiveMutex);
+		m_InactiveCond.wait(guard);
```

Comment:
I guess this should check `Core_IsActive()` again inside the lock?

-[Unknown]

Reply:
All the previous calls to event methods should be revisited to look for bugs, yeah :)